src/Uno.UI/UI/Xaml/VisualStateGroup.cs

```diff
@@ -150,11 +150,15 @@ private void OnParentChanged(object instance, object key, DependencyObjectParent
 			if (this.GetParent() is IFrameworkElement fe)
 			{
 				OnOwnerElementChanged();
+#if !UNO_HAS_ENHANCED_LIFECYCLE
```

Comment:
Why is this needed now? Since it's defined only on wasm, won't it cause issues there?

Reply:
UNO_HAS_ENHANCED_LIFECYCLE is defined on skia and wasm. I moved the StateTriggers initialization to EnterImpl to match WinUI, so we no longer need to subscribe to Loaded to initialize StateTriggers on platforms that fire EnterImpl